win32/src/gballoc_ll_jemalloc.c

```diff
@@ -214,3 +218,240 @@ void gballoc_ll_print_stats(void)
+
+                    if (snprintf_result < 0 || (size_t)snprintf_result >= sizeof(command))
+                    {
+                        /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_018: [ If there are any errors, gballoc_ll_set_option shall fail and return a non-zero value. ]*/
+                        LogError("snprintf(const char* buffer=%p, size_t size=%zu, const char* format=arena.<i>.dirty_decay_ms, i=%" PRIu32 ") failed %s", command, sizeof(command), i, (snprintf_result < 0) ? " with encoding error" : " due to insufficient buffer size");
+                        result = MU_FAILURE;
+                        break;
+                    }
+                    else
+                    {
+                        /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_009: [ gballoc_ll_set_option shall set the dirty decay time for the arena to decay_milliseconds milliseconds by calling je_mallctl with arena.<i>.dirty_decay_ms as the command. ]*/
+                        int arena_decay_result = je_mallctl(command, NULL, NULL, &decay_milliseconds, sizeof(decay_milliseconds));
+                        if (arena_decay_result != 0)
+                        {
+                            if (arena_decay_result == EFAULT)
+                            {
+                                /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_020: [ If je_mallctl returns EFAULT, gballoc_ll_set_option shall continue without failing as this error is expected when the arena doesn't exist. ]*/
+                                LogError("je_mallctl(const char* name=%s, void* oldp=NULL, size_t* oldlenp=NULL, void* newp=%p, size_t newlen=%zu) failed, continuing without failure as the arena doesn't exist, old_decay_milliseconds=%" PRId64 "", command, &decay_milliseconds, sizeof(decay_milliseconds), old_decay_milliseconds);
+                            }
+                            else
+                            {
+                                /*Codes_SRS_GBALLOC_LL_JEMALLOC_28_018: [ If there are any errors, gballoc_ll_set_option shall fail and return a non-zero value. ]*/
+                                LogError("je_mallctl(const char* name=%s, void* oldp=NULL, size_t* oldlenp=NULL, void* newp=%p, size_t newlen=%zu) failed, old_decay_milliseconds=%" PRId64 "", command, &decay_milliseconds, sizeof(decay_milliseconds), old_decay_milliseconds);
+                                result = MU_FAILURE;
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if (i < narenas)
+                {
+                    for (uint32_t j = 0; j < i; j++)
+                    {
+                        (void)snprintf(command, sizeof(command), "arena.%" PRIu32 ".dirty_decay_ms", j);
+                        (void)je_mallctl(command, NULL, NULL, &old_decay_milliseconds, sizeof(old_decay_milliseconds));
```

Comment:
>je_mallctl [](http://example.com/codeflow?start=30&length=10)

undo-ing of previously performed actions should still log if it fails.

Here and everywhere.

Reply:
Done